src/middleware/builtin.ts

```diff
@@ -210,31 +211,41 @@ export function matchConstraints(
  * Middleware that filters out messages that don't match pattern
  */
 export function matchMessage(
-  pattern: string | RegExp,
+  pattern: string | RegExp | (string | RegExp)[],
 ): Middleware<SlackEventMiddlewareArgs<'message' | 'app_mention'>> {
   return async ({ event, context, next }) => {
-    let tempMatches: RegExpMatchArray | null;
-
+    let tempMatches: RegExpMatchArray | null = null;
+    let patternArray: (string | RegExp)[] = [];
     if (!('text' in event) || event.text === undefined) {
       return;
     }
-
-    // Filter out messages or app mentions that don't contain the pattern
-    if (typeof pattern === 'string') {
-      if (!event.text.includes(pattern)) {
-        return;
-      }
+    // Filter out messages or app mentions that don't contain the pattern]
+    if (typeof pattern === 'string' || util.types.isRegExp(pattern)) {
+      patternArray = [pattern];
     } else {
-      tempMatches = event.text.match(pattern);
-
-      if (tempMatches !== null) {
-        context['matches'] = tempMatches;
+      patternArray = pattern;
+    }
+    let matched = false;
+    for (let i = 0; i < patternArray.length; i += 1) {
```

Comment:
code style suggestion: instead of using a `for` loop with the iteration value (`i`), how do you feel about using a more modern syntax? for example, a [`for...of` loop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) is seen as _safer_ than the plain `for` loop. its safer because when you use `for...of`, there's no `i` value which can be unintentionally mutated as the loop is running. it's also cleaner because you never have to index into the original array (never have to worry about an out of bounds error). and it works on any iterable in the language, not just arrays. what do you think?

Reply:
Thanks for the suggestion! I have been looking into this and when I fix the code to use the "for..of.." syntax, I get this error message on running the tests.
"_**iterators/generators require regenerator-runtime, which is too heavyweight for this guide to allow them. Separately, loops should be avoided in favor of array iterations."**_
I agree with everything you said regarding "for..of.." loops; at the same time, they don't seem to work well with the current suite of tests, and I am not sure how to get over that obstacle.